ui/src/model-manager/hooks/useUpdateModels.ts

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useRef, useState } from 'react';
+import { getAllModelsList } from '../../Api';
+import type { ModelsListRespItem } from '../types';
+// @ts-ignore
+import { api } from "/scripts/api.js";
+
+export const useUpdateModels = () => {
+  // all model types
+  const [modelTypeList, setModelTypeList] = useState<string[]>([]);
+
+  // all models
+  const [modelsList, setModelsList] = useState<ModelsListRespItem[]>([]);
+
+  // loading status
+  const [loading, setLoading] = useState(true);
+  const initDataInterval = useRef<number>();
+
+  useEffect(() => {
+    api.addEventListener("download_progress", (e: { detail: [] }) => {
+      if (e.detail.length === 0) { // if download_progress is empty, that means download is done, so re-fetch data
+        setLoading(true); // set loading to true to start interval
+      }
+    });
+  }, []);
+
+  useEffect(() => {
+    if (loading && !initDataInterval.current) { // if loading and interval not started, start interval
+      initDataInterval.current = setInterval(initData, 3000);
+    } else if (!loading && initDataInterval.current) { // if not loading and interval is running, clear interval
+      clearInterval(initDataInterval.current);
+      initDataInterval.current = undefined;
+    }
+
```

Comment:
since you already have ws setup, why not use ws to send message when a new model info is hashed and fetched from civit ai?

Reply:
fixed